drivers/arm/ws2812_pwm.c

```diff
@@ -1 +1,207 @@
-#error("NOT SUPPORTED")
\ No newline at end of file
+#include "ws2812.h"
+#include "quantum.h"
+#include "hal.h"
+
+/* Adapted from https://github.com/joewa/WS2812-LED-Driver_ChibiOS/ */
+
+#ifdef RGBW
+#    error "RGBW not supported"
+#endif
+
+#ifndef WS2812_PWM_DRIVER
+#    define WS2812_PWM_DRIVER PWMD2  // TIMx
+#endif
+#ifndef WS2812_PWM_CHANNEL
+#    define WS2812_PWM_CHANNEL 2  // Channel
+#endif
+#ifndef WS2812_PWM_PAL_MODE
+#    define WS2812_PWM_PAL_MODE 2  // DI Pin's alternate function value
+#endif
+#ifndef WS2812_DMA_STREAM
+#    define WS2812_DMA_STREAM STM32_DMA1_STREAM2  // DMA Stream for TIMx_UP
+#endif
+#ifndef WS2812_DMA_CHANNEL
+#    define WS2812_DMA_CHANNEL 2  // DMA Channel for TIMx_UP
+#endif
+
+#ifndef WS2812_PWM_TARGET_PERIOD
+//#    define WS2812_PWM_TARGET_PERIOD 800000 // Original code is 800k...?
+#    define WS2812_PWM_TARGET_PERIOD 80000  // TODO: work out why 10x less on f303/f4x1
```

Comment:
I have used the same codes for the noah keyboard(stm32f11ce based board), and it should use 800K, not sure why it was modified to 80000.

Reply:
doesnt work with 800K :man_shrugging:

Reply:
Maybe the testing system was running with HSI clock(8mHz, not the HSE at 8x9 on f303 or 8x12 on f411)?